src/main/scala/org/constellation/domain/redownload/RedownloadService.scala

```diff
@@ -147,16 +157,31 @@ class RedownloadService[F[_]](
 
       _ <- if (shouldRedownload(acceptedSnapshots, majorityState, snapshotHeightRedownloadDelayInterval)) {
         val plan = calculateRedownloadPlan(acceptedSnapshots, majorityState)
-        applyRedownloadPlan(plan)
+        applyRedownloadPlan(plan).value // TODO: Avoid mixing F[Unit] with EitherT
       } else F.unit
     } yield ()
 
-  private[redownload] def applyRedownloadPlan(plan: RedownloadPlan): F[Unit] =
+  private[redownload] def applyRedownloadPlan(plan: RedownloadPlan): EitherT[F, Throwable, Unit] =
     for {
-      _ <- fetchAndStoreMissingSnapshots(plan.toDownload)
-      _ <- acceptedSnapshots.modify { m =>
-        val updated = plan.toLeave |+| plan.toDownload
-        (updated, ())
+      _ <- EitherT.liftF(fetchAndStoreMissingSnapshots(plan.toDownload))
+
+      _ <- EitherT.liftF {
+        acceptedSnapshots.modify { m =>
+          val updated = plan.toLeave |+| plan.toDownload
+          (updated, updated)
+        }
+      }
+
+      highestSnapshotInfo <- EitherT
+        .liftF(getAcceptedSnapshots())
+        .map(_.maxBy { case (height, _) => height } match { case (_, hash) => hash })
+        .flatMap(hash => snapshotInfoStorage.readSnapshotInfo(hash))
+
+      _ <- EitherT.liftF(snapshotService.setSnapshot(highestSnapshotInfo))
```

Comment:
each time you're using `liftF`, you're omitting all errors from inside.
it should be `setSnapshot().attemptT` to map error from F/IO into Either/EitherT